src/Validator.php

```diff
@@ -103,4 +103,33 @@ public function validate(string $attribute, $value, array $parameters, Validator
 
         return false;
     }
+    
+    /**
+     * Replace all place-holders for the required_with rule.
+     *
+     * @param  string  $message
+     * @param  string  $attribute
+     * @param  string  $rule
+     * @param  array   $parameters
+     * @return string
+     */
+    public function replacer($message, $attribute, $rule, $parameters, ValidatorContract $validator)
+    {
+        $this->setRequest($validator);// Important to set the request before anything else
```

Comment:
This part is also unnecessary, because the `replacer` method i never called before the `validate` method, which are sets the request.

Reply:
If I remove this and instead dump the request using `dd($this->request);` then I simply get an empty array:
```php
[ ]
```
which will make the `$this->fetchCountryCode($parameter);` fail.

When using this `setRequest()` then dumping `dd($this->request);` returns:
```php
array:2 [▼
  "country" => "AS"
  "postal_code" => "132456789"
]
```

So I am not sure how to get the country from validation otherwise. I am testing with:

```php
\Illuminate\Support\Facades\Validator::make([
        'country' => 'AS',
        'postal_code' => '132456789',
    ], [
        'country' => '',
        'postal_code' => 'postal_code:country',
    ])
```

Reply:
You are right, i was careless to put it that way, it is not even the same instance.
But that means we have to make the replacer specific for this instance, here is a way: 
```
public function validate(string $attribute, $value, array $parameters, ValidatorContract $validator)
    {
        if (!is_string($value) || !$value) {
            return false;
        }

        $this->setRequest($validator);
         //Add the replacer to the validator instance via a closure
        $validator->addReplacer('postal_code', function($message, $attribute, $rule, $parameters) {
            $formats = array();

            foreach ($parameters as $parameter) {
                $countryCode = $this->fetchCountryCode($parameter);
                $formats = array_merge($formats, $this->engine->getFormats($countryCode));
            }

            // Remove any doublicate formats if validating with more than one country
            $formats = array_unique($formats);

            // Implode all formats comma separated
            $format = implode(', ', $formats);

            //Return $message where :format is replaced with comma separated list of formats
            return str_replace([':format'], $format, $message);
        });

        foreach ($parameters as $parameter) {
            if (!$countryCode = $this->fetchCountryCode($parameter)) {
                continue;
            }

            if ($this->engine->isValid($countryCode, $value, true)) {
                return true;
            }
        }

        return false;
    }
```

Reply:
I personally don't feel logic that binds things to the container (or a manager/factory) belongs inside the validator. But we could reuse the validator instance within the `afterResolving` closure in the service provider

simplified example:

```
public function register() {
    $this->app->afterResolving('validator', function(Factory $factory) {
        $postalCodeValidator = $app->make(Validator::class);

        $factory->extend(... $postalCodeValidator);
        $factory->replacer(... $postalCodeValidator);
    });
}
```

Reply:
@axlon  I only posted the code above as one of the ways to overcome this issue, but i agree that taking advantage of the container is much more better